### PR TITLE
Added Meteor Evaluation Metric

### PR DESCRIPTION
**Pull request by CateGitau**
- Added Meteor Evaluation metric
- Made it possible to include more than one evaluation metric for comparison
- Fixed Chrf score function, it was giving out an error.

### CODE REVIEW

File: joeynmt/training.py [hunk 1]
```diff
@@ -103,11 +108,12 @@ def __init__(self, model: Model, config: dict) -> None:
         if self.early_stopping_metric in ["ppl", "loss"]:
             self.minimize_metric = True
         elif self.early_stopping_metric == "eval_metric":
-            if self.eval_metric in ["bleu", "chrf"]:
-                self.minimize_metric = False
-            # eval metric that has to get minimized (not yet implemented)
-            else:
-                self.minimize_metric = True
+            for eval_metric in self.eval_metrics:
```
Comment (juliakreutzer) [inline joeynmt/training.py:111]: This is ambiguous: for early stopping, we actually have to pick one of the metrics, and minimize_metric has to be set according to that one. So I propose we take by default the first evaluation metric. That would just need documentation and instead of looping over the metrics here, we would only take the first.

File: joeynmt/training.py [hunk 2]
```diff
@@ -391,7 +398,7 @@ def train_and_validate(self, train_data: Dataset, valid_data: Dataset) \
                     elif self.early_stopping_metric in ["ppl", "perplexity"]:
                         ckpt_score = valid_ppl
                     else:
-                        ckpt_score = valid_score
+                        ckpt_score = valid_scores[0]
```
Comment (juliakreutzer) [inline joeynmt/training.py:401]: Okay, so you're already using the first metric as the early stopping one. Good :)

File: joeynmt/training.py [hunk 3]
```diff
@@ -410,10 +417,11 @@ def train_and_validate(self, train_data: Dataset, valid_data: Dataset) \
                         self.scheduler.step(ckpt_score)
 
                     # append to validation report
-                    self._add_report(
-                        valid_score=valid_score, valid_loss=valid_loss,
-                        valid_ppl=valid_ppl, eval_metric=self.eval_metric,
-                        new_best=new_best)
+                    for idx, valid_score in enumerate(valid_scores):
+                        self._add_report(
```
Comment (juliakreutzer) [inline joeynmt/training.py:421]: I'd rather pass multiple validation scores to _add_report to add more columns rather than rows to the validation score file.

File: joeynmt/training.py [hunk 4]
```diff
@@ -425,12 +433,13 @@ def train_and_validate(self, train_data: Dataset, valid_data: Dataset) \
...
-                        'Validation result (greedy) at epoch %3d, '
-                        'step %8d: %s: %6.2f, loss: %8.4f, ppl: %8.4f, '
-                        'duration: %.4fs', epoch_no + 1, self.steps,
-                        self.eval_metric, valid_score, valid_loss,
-                        valid_ppl, valid_duration)
+                    for idx, valid_score in enumerate(valid_scores):
+                        self.logger.info(
+                            'Validation result (greedy) at epoch %3d, '
+                            'step %8d: %s: %6.2f, loss: %8.4f, ppl: %8.4f, '
+                            'duration: %.4fs', epoch_no + 1, self.steps,
```
Comment (juliakreutzer) [inline joeynmt/training.py:440]: Same here, just add all the metrics to the info string, and not repeat logging the stats.

**Review by juliakreutzer — CHANGES_REQUESTED** [review]
Thanks @CateGitau !! Sorry it took so long for me to get to review this. There are a few changes I'd like to see to make it consistent (see review), but overall it's a good solution!
